crates/federated-server/src/server.rs

```diff
@@ -193,13 +194,30 @@ pub async fn router<R: engine::Runtime, SR: ServerRuntime, H: HooksExtension>(
 
     let ct = match &config.mcp {
         Some(mcp_config) if mcp_config.enabled => {
-            let (mcp_router, ct) = grafbase_mcp::router(engine, mcp_config);
+            let (mcp_router, ct) = grafbase_mcp::router(&engine, mcp_config);
             router = router.merge(mcp_router);
             ct
         }
         _ => None,
     };
 
+    for public_metadata_endpoint in engine
```

Comment:
you should retrieve directly through the extension, not the engine-auth struff. only there to handle the legacy jwt provider. unless I'm missin something